src/rocket/introspection/guard.rs

```diff
@@ -154,29 +154,33 @@ impl<'request> FromRequest<'request> for &'request IntrospectedUser {
 impl<'a> OpenApiFromRequest<'a> for &'a IntrospectedUser {
     fn from_request_input(
         _gen: &mut OpenApiGenerator,
-        name: String,
-        required: bool,
+        _name: String,
+        _required: bool,
+        request: &Request,
     ) -> rocket_okapi::Result<RequestHeaderInput> {
+        // Setup global requirement for Security scheme
         let security_scheme = SecurityScheme {
-            data: rocket_okapi::okapi::openapi3::SecuritySchemeData::Http {
-                scheme: "bearer".to_string(),
-                bearer_format: None,
+            description: Some(
+                "Use OpenID Connect to authenticate. (does not work in RapiDoc at all)".to_owned(),
+            ),
+            data: SecuritySchemeData::OpenIdConnect {
+                open_id_connect_url: "https://auth.domain.com/.well-known/openid-configuration"
```

Comment:
help needed
how to make the url dynamic with the config?

Reply:
as far as I know, you should be able to somehow access the rocket object and fetch the configuration for it. Like what I do in [rocket/introspection/guard.rs](https://github.com/smartive/zitadel-rust/blob/main/src/rocket/introspection/guard.rs#L77). The rocket call should be available on the request object. You can use the same logic and have some OpenApiConfig or something similar injected in the state and then fetch it from the request during our "from request" call. (if that works)

Reply:
@buehler i just fixxed it with the newest commit, but i need to use the Rocket.toml for that

Reply:
Hey @NewtTheWolf 
Why do you need the rocket.toml?

While I like the possibility of having the configuration in the rocket.toml, a programmatic way should exist as well. Wdyt?

Reply:
@buehler With the knowledge I currently have, there are three options:

1. A custom Config Loader
2. `Rocket.toml`
3. Environment variables

As shown here it won't be possible: https://github.com/smartive/zitadel-rust/blob/1fc35131b7e61e3e98db23c1cb907fd0382b30d3/src/rocket/introspection/guard.rs#L77

because I don't know how to get the request object to retrieve the state, and I don't know any other way to access the state.

Therefore, the library itself could either set an environment variable or use its own config loader (or the `Rocket.toml`).

And I think using the `Rocket.toml` in general (where possible) would be a great idea because the `Rocket.toml` is designed for configuring your Rocket server.